### PR TITLE
Update infrastructure (actions)

### LINKED ISSUE #161: update branch of gh-action-pypi-publish
https://github.com/pysal/mapclassify/blob/130a1cfb80bbc2544e429e1820acfae155ababa3/.github/workflows/release.yml#L42

xref #160, https://github.com/pysal/spopt/issues/337

--> `pypa/gh-action-pypi-publish@release/v1`

### LINKED ISSUE #162: GHA for updating `versioneer`
xref https://github.com/pysal/spopt/pull/340, #161, #160

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR:
* resolves #162
  *  this is [passing](https://github.com/jGaboardi/mapclassify/pull/2) 
* resolves #161

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -73,7 +72,7 @@ def setup_package():
     setup(
         name=package,
         version=versioneer.get_version(),
-        cmdclass=versioneer.get_cmdclass({"build_py": build_py}),
```
Comment (martinfleis) [inline setup.py:76]: Don't remember why we had this, any clue?

Comment (jGaboardi) [reply setup.py:76]: May have been legacy PySAL stuff that we kept around when transitioning to `versioneer`: https://github.com/pysal/libpysal/pull/429

Comment (jGaboardi) [reply setup.py:76]: It was giving me trouble over in `spopt` so I tried [getting rid of it](https://github.com/pysal/spopt/pull/340/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7L37). Passed after that.

Comment (martinfleis) [reply setup.py:76]: okay :D

**Pull request merged**